3515/1/top_gems.rb

```diff
@@ -0,0 +1,48 @@
+require 'yaml'
+require 'gems'
+require 'mechanize'
+require 'httparty'
+require 'terminal-table'
+require 'pry'
+require './files/terminal_table'
+require './files/gems_get_params'
+require './files/gems_loader'
+require './files/url_loader'
+
+class GemsMain
+  attr_reader :gems_table_arr
+  attr_reader :threads
+  attr_reader :gems_arr
+
+  def main_process
+    gems_arr.gems_arr.each do |gem_name|
+      threads << Thread.new(gem_name) do
+        gem_url = GemsUrlLoader.new(gem_name)
+        next unless gem_url.url.include?('http')
+
+        aaa(gem_url.url, gem_name)
+      end
+    end
+  end
+
+  def aaa(gem_url, gem_name)
+    gem_params = GemParamGetter.new(gem_url, gem_name)
+    gems_table_arr << gem_params.params
+  end
+
+  def gems_main
+    gems_arrr
+    main_process
+    threads.each(&:join)
+    TerminalTable.new(gems_table_arr)
+  end
+
+  def gems_arrr
+    @gems_arr = GemsLoader.new
+    @gems_table_arr = []
+    @threads = []
+  end
+end
+
+a = GemsMain.new
+a.gems_main
```

Comment:
What do you do by calling that method? It should be clear from method call what it does. And be attentive to correct variables naming